### PR TITLE
feat: Adding publishing result as CSV to Azure capability

### PR DESCRIPTION
**Pull request by dennisreis-tomtom**

### CODE REVIEW

File: azure_function/resource-graph-collector/__init__.py [hunk 1]
```diff
@@ -44,19 +48,34 @@ def main(event: func.TimerRequest) -> None:
...
             loki_logger.publish(item, fields_to_labels)
+
+        # publish to Azure Blob container
+        azure_publisher = AzurePublisher(
+            connection_string=os.environ["STORAGE_ACCOUNT_CONNECTION"],
+            container_name=os.environ["CONTAINER_NAME"],
+        )
+        azure_publisher.publish(
+            name=resource_graph_query.name, data=result.as_dict().get("data", [])
+        )
```
Comment (GiuseppeChiesa-TomTom) [inline azure_function/resource-graph-collector/__init__.py]: could you add this behind a feature switch (env var is ok)? so it's backward compatible

**Review by GiuseppeChiesa-TomTom — CHANGES_REQUESTED** [review]
nice addition, please check my improvement proposal

File: azure_function/resource-graph-collector/__init__.py (hunk 1 above)
Comment (dennisreis-tomtom) [reply azure_function/resource-graph-collector/__init__.py]: Great idea, I've push a new proposal. Looking forwarding hearing from you.

File: azure_function/resource-graph-collector/publisher.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[publisher]
+loki=True
+azure=False
```
Comment (gchiesa) [inline azure_function/resource-graph-collector/publisher.properties]: it's not clear to me how this is going to work. You plan to pass this file during the deployment in terraform?
I think we can make it much simpler, by adding 2 env vars:
`ENABLE_LOKI_PUBLISHER=true|false` and `ENABLE_AZURE_BLOB_PUBLISHER=true|false` 
and we drive the switches in the function app configuration. 
So we don't need a static file with config parse.

What you think?

**Review by gchiesa — CHANGES_REQUESTED** [review]

File: azure_function/resource-graph-collector/publisher.properties (hunk 2 above)
Comment (dennisreis-tomtom) [reply azure_function/resource-graph-collector/publisher.properties]: Indeed the env vars are way more simples. My initial tough was having this file so it's just one single point of config for all possible backend, although I do agree that we don't need that now. 
And yes the idea is to pass this file during TF deployment. Anyway I'm changing for the env var solution :), thanks for the advise

File: azure_function/resource-graph-collector/__init__.py [hunk 3]
```diff
@@ -44,19 +60,36 @@ def main(event: func.TimerRequest) -> None:
...
-        if fields_to_labels_string:
-            fields_to_labels = [e.strip() for e in fields_to_labels_string.split(',')]
-
-        if len(fields_to_labels) > MAX_LABELS:
-            raise ValueError(f"A maximum of {MAX_LABELS} is supported. You requested: {len(fields_to_labels)} labels, "
-                            f"namely: [{', '.join(fields_to_labels)}]")
-
-        for item in result.as_dict().get('data', []):
-            loki_logger.publish(item, fields_to_labels)
+        if is_enabled(os.environ.get("ENABLE_LOKI_PUBLISHER", "true")):
```
Comment (dennisreis-tomtom) [inline azure_function/resource-graph-collector/__init__.py:63]: I chose this approach having the backward compatibility in mind. So even if they inform the new environment variable the default value will be used, which I've deliberately chose "true" for Loki and "false" for Azure. WDYT?

Comment (GiuseppeChiesa-TomTom) [reply azure_function/resource-graph-collector/__init__.py:63]: LGTM

File: main.tf [hunk 4]
```diff
@@ -61,12 +61,17 @@ resource "azurerm_function_app" "function_app" {
...
+    "USER_ASSIGNED_IDENTITY_APP_ID"         = data.azurerm_user_assigned_identity.identity.client_id
+    "LOKI_USERNAME"                         = var.loki_authentication.username
+    "LOKI_PASSWORD"                         = var.loki_authentication.password
+    "LOKI_ENDPOINT"                         = var.loki_endpoint_url
+    "LOKI_LABEL_NAMES"                      = var.loki_label_names
+    "RESOURCE_GRAPH_QUERY_IDS"              = var.resource_graph_query_ids
+    "STORAGE_ACCOUNT_CONNECTION"            = azurerm_storage_account.sa.primary_connection_string
+    "CONTAINER_NAME"                        = azurerm_storage_container.sacontainer.name
+    "ENABLE_LOKI_PUBLISHER" : var.enable_loki_publisher
+    "ENABLE_AZURE_BLOB_PUBLISHER" : var.enable_azure_blob_publisher
```
Comment (GiuseppeChiesa-TomTom) [inline main.tf]: will this work with `:` instead `=`?

**Review by GiuseppeChiesa-TomTom — COMMENTED** [review]
please check my comment. I'm not sure about the reason to use `:` instead extending the map with `=` sign.

File: main.tf (hunk 4 above)
Comment (dennisreis-tomtom) [reply main.tf]: Wow, indeed this is a typo although I've deployed this version yesterday in the dev subscription and it's running ok :smile: . Nevertheless, I'm changing it right now, thanks for the heads-up

**Review by gchiesa — APPROVED** [review]

**Pull request merged**